### ISSUE TITLE
Custom filtering

### ISSUE THREAD

**Issue by thomasfl** [angular-ui/ui-grid#165]
I can't find any examples on how to write custom filters for ng-grid. It's easy to write a custom filter that filters out rows from ng-grid by simply manipulating the data used by ng-grid, but I have a fairly large dataset and don't want to keep an extra copy of the data. 

Is it possible to use the gridOptions.useExternalFilter option? I can't find any examples the explains what it does.

Here's a very simple plunk 
http://plnkr.co/edit/T6qaQX

**Comment by lecoleco** [comment]
When  make a custom filter the selection not work correct. If select the first row and put a text that does not exist on the grid after clear the input the selected item not show anymore and after put same text like information has on the second page the first item on second page show selected. I use state! Existe another way to do that?

**Comment by tmpathak-cygnet** [comment]
I have used same filter for the filtering the Data but filtering is very slow is there any way to make it more fast ??

**Comment by kalyanidesai** [comment]
Hello I am using Angular UI Grid to display the data in neat and clean manner. In the UI grid, there is one column as "Duration". I want to sort this column in descending order but unfortunately, i am unable to do that. I tried but it is sorting in text format.
In the controller part I have written as follows:
$scope.sortTrendingEvents = function () {
$scope.slaGridOptions.data = $filter('orderBy')($scope.slaGridOptions.data, "duration", true)
console.log('this button is clicked when trending event is pressed');
};
In the HTML part I have written as:

Trending Events
In short what I have done is on the click of a button the Duration column must be sorted in DESC order but with the help of this code,

I am unable to sort this column in number pattern.
Please do guide me. Thank you so much.
